applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -642,9 +658,30 @@ impl AppStateInner {
     }
 
     pub async fn refresh_balance(&mut self) -> Result<(), UiError> {
-        let balance = self.wallet.output_manager_service.get_balance().await?;
-        self.data.balance = balance;
-        self.updated = true;
+        if Instant::now().duration_since(self.balance_enquiry_cooldown_start) >
+            Duration::from_secs(self.balance_enquiry_cooldown_period)
+        {
+            self.balance_enquiry_cooldown_start = Instant::now();
+            self.balance_enquiry_cooldown_infringement = false;
+            let balance = self.wallet.output_manager_service.get_balance().await?;
+            self.data.balance = balance;
+            self.updated = true;
+        } else {
+            self.balance_enquiry_cooldown_infringement = true;
+        }
+
+        Ok(())
+    }
+
+    pub async fn refresh_balance_check(&mut self) -> Result<(), UiError> {
```

Comment:
I am struggling to understand the purpose of this?

Reply:
Please see above

Reply:
This can be removed. See above